File: examples/c++/table_row_title.cpp
#include <string>

class TableRowTitle {
public:
    std::string text;
    std::string alignment = "left";
    int rowIndex = 0;
    bool isHeader = false;
    void setText(const std::string& t) { text = t; }
    void setAlignment(const std::string& a) { alignment = a; }
    std::string render() { return text; }
    void markAsHeader() { isHeader = true; }
};

File: examples/python/table_row_title.py
import html as _html


class TableRowTitle:
    def __init__(self, text="", row_index=0):
        self.text = text
        self.alignment = "left"
        self.row_index = row_index
        self.is_header = False

    def set_text(self, t):
        self.text = t

    def set_alignment(self, a):
        self.alignment = a

    def render(self):
        tag = "th" if self.is_header else "td"
        return f"<{tag} style=\"text-align:{self.alignment}\">{_html.escape(self.text)}</{tag}>"

    def mark_as_header(self):
        self.is_header = True
